docarray/array/storage/milvus/getsetdel.py

```diff
@@ -1,69 +1,66 @@
 from docarray.array.storage.base.getsetdel import BaseGetSetDelMixin
 from docarray import Document
 from docarray.array.storage.base.helper import Offset2ID
-from docarray.array.storage.milvus.backend import always_true_expr
+from docarray.array.storage.milvus.backend import always_true_expr, ids_to_milvus_expr
 
 
 class GetSetDelMixin(BaseGetSetDelMixin):
     def _get_doc_by_id(self, _id: str) -> 'Document':
         # to be implemented
-        self._get_docs_by_ids([_id])
+        return self._get_docs_by_ids([_id])
 
     def _del_doc_by_id(self, _id: str):
         # to be implemented
         self._del_docs_by_ids([_id])
 
     def _set_doc_by_id(self, _id: str, value: 'Document'):
         # to be implemented
-        self._set_doc_by_id([_id], [value], None)
+        self._set_docs_by_ids([_id], [value], None)
 
     def _load_offset2ids(self):
-        collection = self._offset2id_collection  # Get an existing collection.
+        collection = self._offset2id_collection
         collection.load()
         res = collection.query(
-            expr='(document_id in ["1"]) or (document_id not in ["1"])',  # is this the correct expr to say "all"?
+            expr='(document_id in ["1"]) or (document_id not in ["1"])',
             # output_fields=["book_id", "book_intro"],
             consistency_level="Strong",
         )
         collection.release()
         sorted_res = sorted(res, key=lambda k: int(k['offset']))
-        self._offset2id = Offset2ID(r['document_id'] for r in sorted_res)
+        self._offset2ids = Offset2ID([r['document_id'] for r in sorted_res])
 
     def _save_offset2ids(self):
         collection = self._offset2id_collection  # Get an existing collection.
         # delete old entries
         collection.delete(
-            expr=always_true_expr(
-                'document_id'
-            ),  # is this the correct expr to say "all"?
-            consistency_level="Strong",
+            expr=always_true_expr('document_id'),
+            consistency_level='Strong',
         )
         # insert current entries
-        ids = self._offset2id.ids
+        ids = self._offset2ids.ids
         offsets = [str(i) for i in range(len(ids))]
         collection.insert([offsets, ids])
 
     def _get_docs_by_ids(self, ids: 'Iterable[str]') -> 'DocumentArray':
-        id_list_str = '[' + ','.join(ids) + ']'
+        self._collection.load()
         res = self._collection.query(
-            expr=f'document_id in {id_list_str}',
-            # output_fields=["book_id", "book_intro"],
+            expr=f'document_id in {ids_to_milvus_expr(ids)}',
+            output_fields=['serialized'],
             consistency_level='Strong',
         )
-        # TODO(johannes) handle output an convert to da
+        self._collection.release()
```

Comment:
this looks like something that would require a context manager usage

Reply:
you from their side, or our `with da`?

I was already thinking about putting this in our context manager, but that would tie this memory loading/releasing to the updating of offset2id. I /think/ that is what we want, but we need to think about corner cases.

I think I will keep it like this for now in order to make fast progress on all the features that we have to support, but we have to keep this in mind for sure.

Reply:
I was saying from their side? They do not offer this as ctxt manager? if not, we need to do the `try: finally` manually

Reply:
No they don't. I think the following is the best course of action, also in order to avoid loading and releasing constantly when it is not necessary:

- we have an internal context manager that does the load and release wherever it is needed; basically just wrapping their load and release
- the `with da` also does the load and release, so that we can avoid loading and releasing at every operation
- If the we are in the scope of `with da`, the internal context manager does nothing, since `with da` handles it; if we are outside of such a scope, the internal context manager does it

Is this overkill?